Language/Java/Transformer.hs

```diff
@@ -1,173 +1,286 @@
+      (transformToAnalyzed (addLocalVarIdents [idnt] scope) stmt)
+  transformToAnalyzed _ Empty = Empty
+  transformToAnalyzed scope (ExpStmt srcspan expr) = ExpStmt srcspan (transformToAnalyzed scope expr)
+  transformToAnalyzed scope (Assert expr mbExp) = Assert (transformToAnalyzed scope expr) (fmap (transformToAnalyzed scope) mbExp)
+  transformToAnalyzed scope (Switch switchStyle expr switchBlocks) = Switch switchStyle (transformToAnalyzed scope expr) (map (transformToAnalyzed scope) switchBlocks)
+  transformToAnalyzed scope (Do srcspan expr stmt) = Do srcspan (transformToAnalyzed scope expr) (transformToAnalyzed scope stmt)
+  transformToAnalyzed _ (Break srcspan mbIdent) = Break srcspan mbIdent
+  transformToAnalyzed _ (Continue mbIdent) = Continue mbIdent
+  transformToAnalyzed scope (Return srcspan mbExp) = Return srcspan (fmap (transformToAnalyzed scope) mbExp)
+  transformToAnalyzed scope (Synchronized expr block) = Synchronized (transformToAnalyzed scope expr) (transformToAnalyzed scope block)
+  transformToAnalyzed scope (Throw expr) = Throw (transformToAnalyzed scope expr)
+  transformToAnalyzed scope (Try srcspan tryResources block catches mbBlock) =
     Try
       srcspan
-      (map (analyze scope) tryResources)
-      (analyze (addLocalVars scope (localVarsInTryResources tryResources)) block)
-      (map (analyze scope) catches)
-      (fmap (analyze scope) mbBlock)
-  analyze scope (Labeled idnt stmt) = Labeled idnt (analyze scope stmt)
-
--- classification of MethodInvocation name
-instance Transform MethodInvocation where
-  -- Todo : classify multipart name
-  analyze scope (MethodCall Nothing idnt args) = MethodCall Nothing idnt (map (analyze scope) args)
-  analyze scope (MethodCall (Just name@(Name _ [singleId])) idnt args) = MethodCall (Just (classifyName singleId scope name)) idnt (map (analyze scope) args)
-  analyze scope (MethodCall _ idnt args) = MethodCall Nothing idnt (map (analyze scope) args)
-  analyze scope (PrimaryMethodCall expr refTypes idnt args) = PrimaryMethodCall (analyze scope expr) refTypes idnt (map (analyze scope) args)
-  analyze scope (SuperMethodCall refTypes idnt args) = SuperMethodCall refTypes idnt (map (analyze scope) args)
-  analyze scope (ClassMethodCall name refTypes idnt args) = ClassMethodCall name refTypes idnt (map (analyze scope) args)
-  analyze scope (TypeMethodCall name refTypes idnt args) = TypeMethodCall name refTypes idnt (map (analyze scope) args)
-
-classifyName :: Ident -> IdentCollection -> Name -> ClassifiedName
-classifyName idnt ic =
-  if isIdent idnt ic then ExpressionName else TypeName
+      (map (transformToAnalyzed scope) tryResources)
+      (transformToAnalyzed (addLocalVarIdents (localVarIdentsFromTryResources tryResources) scope) block)
+      (map (transformToAnalyzed scope) catches)
+      (fmap (transformToAnalyzed scope) mbBlock)
+  transformToAnalyzed scope (Labeled idnt stmt) = Labeled idnt (transformToAnalyzed scope stmt)
 
 -- helper functions
```

Comment:
Die Hilfsfunktionen sollten abstrahiert genug in ein anderes passendes Modul ausgelagert werden.